tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -17,32 +17,46 @@ var _ = Describe("odo devfile push command tests", func() {
 	var namespace, context, cmpName, currentWorkingDirectory, projectDirPath string
 	var projectDir = "/projectDir"
 	var sourcePath = "/projects/nodejs-web-app"
+	var clusterType helper.CliRunner
 
-	// TODO: all oc commands in all devfile related test should get replaced by kubectl
-	// TODO: to goal is not to use "oc"
-	oc := helper.NewOcRunner("oc")
+	// Using program commmand according to cluter type in devfile
+	if os.Getenv("KUBERNETES") == "true" {
```

Comment:
This condition should be extracted into a helper. Something like getCliRunner

Reply:
> This condition should be extracted into a helper. Something like getCliRunner

This is just a env variable I am fetching it over here. It has nothing to deal with like the name you suggest `getCliRunner`.

Reply:
I don’t think we are worried about over crowding the helper file 🙂

Reply:
I see +3 instances of this condition, that’s 15 lines of code we can reduce by adding a small function, I hope that’s a good enough trade off to have to function in the helper file?

Reply:
@girishramnani Yes, we should have it in a function, exactly same we are already doing in pr  [here](https://github.com/openshift/odo/pull/3221/files#diff-c5e2055f719fdb07a45e8abeac3b112eR207). I don't see any point adding it in this pr.

Reply:
Think about it this way, in between merging that PR and your PR there will be a time difference where who so ever adds a new test file will use this same if condition instead of the helper function. And then there will be another PR after a while which would clean that up - if we remember that is.
So please do the change even if it’s redundant. It’s a good cleanup.

Reply:
ok, Anyway the [pr](https://github.com/openshift/odo/pull/3221) will cleanup it again. As part of helper file over crowding is concern, i will think later how to split the file on the basis of function we use there if volume of helper increases.

Reply:
Done